test/object-store/sync/flx_sync.cpp

```diff
@@ -4791,6 +4791,255 @@ TEST_CASE("flx: pause and resume bootstrapping at query version 0", "[sync][flx]
+                                 {"firstName", PropertyType::String},
+                                 {"lastName", PropertyType::String}}}};
+
+    size_t num_emps = 5000;
+    size_t num_mgrs = 200;
+    size_t num_dirs = 25;
+    size_t num_total = num_emps + num_mgrs + num_dirs;
+    auto fill_person_schema = [num_emps, num_mgrs, num_dirs](SharedRealm realm) {
+        size_t cnt = 1;
+        {
+            CppContext c(realm);
+            for (size_t i = 0; i < num_mgrs; ++i) {
+                auto obj = Object::create(c, realm, "Person",
+                                          std::any(AnyDict{
+                                              {"_id", ObjectId::gen()},
+                                              {"age", static_cast<int64_t>(40 + i)},
+                                              {"role", "manager"s},
+                                              {"firstName", util::format("firstname-%1", cnt)},
+                                              {"lastName", util::format("lastname-%1", cnt)},
+                                          }));
+                ++cnt;
+            }
+        }
+        {
+            CppContext c(realm);
+            for (size_t i = 0; i < num_emps; ++i) {
+                auto obj = Object::create(c, realm, "Person",
+                                          std::any(AnyDict{{"_id", ObjectId::gen()},
+                                                           {"age", static_cast<int64_t>(20 + i % 30)},
+                                                           {"role", "employee"s},
+                                                           {"firstName", util::format("firstname-%1", cnt)},
+                                                           {"lastName", util::format("lastname-%1", cnt)}}));
+                ++cnt;
+            }
+        }
+        {
+            CppContext c(realm);
+            for (size_t i = 0; i < num_dirs; ++i) {
+                auto obj = Object::create(c, realm, "Person",
+                                          std::any(AnyDict{{"_id", ObjectId::gen()},
+                                                           {"age", static_cast<int64_t>(45 + i)},
+                                                           {"role", "director"s},
+                                                           {"firstName", util::format("firstname-9999%1", cnt)},
+                                                           {"lastName", util::format("lastname-9999%1", cnt)}}));
+                ++cnt;
+            }
+        }
+    };
+
+    enum TestState { initial, disconnected, connecting, connected, bs_downloading, bs_downloaded, bs_complete };
+    TestingStateMachine<TestState> machina(initial);
+
+    auto logger = util::Logger::get_default_logger();
+    FLXSyncTestHarness harness("flx_role_change_bootstrap", {person_schema, {"role", "firstName", "lastName"}});
+    auto& app_session = harness.session().app_session();
+    // Enable the role change bootstraps
```

Comment:
Is that because this depends on https://jira.mongodb.org/browse/BAAS-28604, or something else?
If so, can you reference that in here like `TODO (BAAS-28604): ...` and be sure to link that ticket to RCORE-1872?

Reply:
Yes - this code depends on using the feature flag instead of the protocol version for now. Once the feature is released, the client and server will coordinate the version bump when it will be merged from the feature branch to master. I added a comment to reflect this.